contracts/base/Dispatcher.sol

```diff
@@ -188,6 +200,7 @@ abstract contract Dispatcher is Payments, V2SwapRouter, V3SwapRouter, Callbacks
             abi.decode(inputs, (bytes, address, address, uint256, address));
 
         // this approval is auto-wiped when the token is transferred
+        // TODO: can we save gas here by not approving if router already owns the NFT? i.e. ERC20 -> NFT -> ETH
```

Comment:
confused by this, we're approving here so that an external protocol can pull the NFT from us (the router), so I would think it's necessary

Reply:
Yeah hmm the comment doesnt make sense, not sure what I was referring to.